enhancements/node-pull-credentials/pull-credentials.md

```diff
@@ -0,0 +1,236 @@
+
+## Summary
+
+Allow OpenShift users to import and use images from any registry configured
+during or after the cluster installation by sharing node's pull credentials
+with `openshift-api`, `builder` and `image-registry` pods.
+
+## Motivation
+
+To have image stream, image registry and builds working closely to node avoids
+complexity and redundancy as each component won't need to mount different
+Secrets to have access to information that is already present on the node's
+filesystem.
+
+OpenShift should seamlessly use the cluster-wide pull secret provided during
+installation in builds, imagestream imports and pull-through operations. This
+is particularly important for images pulled from `registry.redhat.io`, which
+requires a pull secret.
+
+Today pull secrets provided during cluster install are available on the
+node's filesystem. If user attempts to import an image stream or pull-through
+from these locations, OpenShift fails as none of `openshift-api`, `builder` or
+`image-registry` use the credentials provided during the installation.
+
+### Goals
+
+- Allow users to use container images from any registry provided during or
+  after cluster installation.
+- Allow users to run `builds` based on images hosted in any registry provided
+  during or after cluster installation without providing any extra credentials.
+- Allow `image-registry` to execute pull-through using the node's pull
+  credentials.
+
+### Non-Goals
+
+- Change the way users manage cluster pull credentials.
+  * Additional credentials may still be provided through secrets on namespace.
+- Use Node mirroring config.
+  * To use config files files such as `/etc/containers/registries.conf` is not
+  part of this proposal.
+- Use Node CAs.
+  * This should be done in the future through another enhancement proposal as
+  we currently have different sources of truth when it comes to CAs (different
+  methods of providing them to different parts of the codebase).
+
+## Proposal
+
+### User Stories
+
+As an OpenShift user
+I want to be able to import imagestreams from any image registry that the
+cluster nodes can pull images from
+so that I can use them without manually creating any extra credentials.
+
+As an OpenShift user
+I want to be able to use, during builds, images from any image registry that
+the cluster nodes can pull images from
+so that I can use these images as base for my own images.
+
+As an OpenShift user
+I want to be able to pull-through images from any registry that the cluster
+nodes can pull images from
+so that I can use image registry cache to speed-up my builds.
+
+### Implementation Details/Notes/Constraints
+
+This is the node's filesystem path we are taking into account:
+
+- `/var/lib/kubelet/config.json` contains the node's pull secret.
+
+#### Image Stream Import
+
+- Mount the pull secret as `readOnly`, `hostPathreadOnly` in
+  `openshift-apiserver` deployment under `/node/var/lib/kubelet/config.json`.
```

Comment:
@dmage we need to come up with an agreement when it comes to the place where we are going to mount this. What is your idea?

Reply:
my idea is to use the same path inside the pod as on the node, but I'd like to hear ideas from someone who can approve PRs for openshift-apiserver.

Reply:
and I'd say it should be configurable, i.e. the operator should mount `/var/lib/kubelet/config.json` and pass this name into the configuration file.

Reply:
@csrwng we would also need to track mounting this into the hypershift-toolkit.

Reply:
Ack

Reply:
Following @dmage suggestion I have updated the proposal, now we mount pull secrets on `/var/lib/kubelet/config.json`